components/properties/src/bidi.rs

```diff
@@ -2,216 +2,68 @@
-    /// assert_eq!(close_paren.mirrored, true);
-    /// let open_angle_bracket = bidi_data.get32_mirroring_props('<' as u32);
-    /// assert_eq!(open_angle_bracket.mirroring_glyph, Some('>'));
-    /// assert_eq!(open_angle_bracket.mirrored, true);
-    /// let close_angle_bracket = bidi_data.get32_mirroring_props('>' as u32);
-    /// assert_eq!(close_angle_bracket.mirroring_glyph, Some('<'));
-    /// assert_eq!(close_angle_bracket.mirrored, true);
-    /// let three = bidi_data.get32_mirroring_props('3' as u32);
-    /// assert_eq!(three.mirroring_glyph, None);
-    /// assert_eq!(three.mirrored, false);
-    /// ```
-    pub fn get32_mirroring_props(&self, code_point: u32) -> BidiMirroringProperties {
-        let bidi_aux_props = self.data.trie.get32(code_point);
-        let mirroring_glyph_opt =
-            Self::convert_mirroring_glyph_data(bidi_aux_props.mirroring_glyph);
-        BidiMirroringProperties {
-            mirroring_glyph: mirroring_glyph_opt,
-            mirrored: bidi_aux_props.mirrored,
-        }
+    fn to_unaligned(self) -> Self::ULE {
+        let [a, b, c, _] = TrieValue::to_u32(self).to_le_bytes();
+        RawBytesULE([a, b, c])
     }
-
-    /// Return a struct for the given code point representing Bidi bracket
-    /// pairing-related property values. See [`BidiPairingProperties`]
-    ///
-    /// # Examples
-    /// ```
-    /// use icu::properties::bidi::{BidiAuxiliaryProperties, BidiPairingProperties};
-    ///
-    /// let bidi_data = BidiAuxiliaryProperties::new();
-    ///
-    /// let open_paren = bidi_data.get32_pairing_props('(' as u32);
-    /// assert_eq!(open_paren, BidiPairingProperties::Open(')'));
-    /// let close_paren = bidi_data.get32_pairing_props(')' as u32);
-    /// assert_eq!(close_paren, BidiPairingProperties::Close('('));
-    /// let open_angle_bracket = bidi_data.get32_pairing_props('<' as u32);
-    /// assert_eq!(open_angle_bracket, BidiPairingProperties::None);
-    /// let close_angle_bracket = bidi_data.get32_pairing_props('>' as u32);
-    /// assert_eq!(close_angle_bracket, BidiPairingProperties::None);
-    /// let three = bidi_data.get32_pairing_props('3' as u32);
-    /// assert_eq!(three, BidiPairingProperties::None);
-    /// ```
-    pub fn get32_pairing_props(&self, code_point: u32) -> BidiPairingProperties {
-        let bidi_aux_props = self.data.trie.get32(code_point);
-        let mirroring_glyph = bidi_aux_props.mirroring_glyph;
-        let paired_bracket_type = bidi_aux_props.paired_bracket_type;
-        match paired_bracket_type {
-            CheckedBidiPairedBracketType::Open => BidiPairingProperties::Open(mirroring_glyph),
-            CheckedBidiPairedBracketType::Close => BidiPairingProperties::Close(mirroring_glyph),
-            _ => BidiPairingProperties::None,
-        }
+    fn from_unaligned(unaligned: Self::ULE) -> Self {
+        let [a, b, c] = unaligned.0;
+        TrieValue::try_from_u32(u32::from_le_bytes([a, b, c, 0])).unwrap_or_default()
```

Comment:
hm. a non-safety invariant of ZeroVec is that byte equality is equality, are we okay with giving that up here?

The worst that can happen is that `==` on `ZeroVec` will do unexpected things with bad data, but I'm overall fine with that. Either way, I think we should document that this breaks the invariant but is convenient

cc @sffc

Reply:
Why do you think this is being given up here?

Reply:
The `unwrap_or_default()` means that two unequal byte sequences will produce equal values